service/src/test/java/uk/gov/hmcts/reform/fpl/service/InboxLookupServiceTest.java

```diff
@@ -0,0 +1,108 @@
+package uk.gov.hmcts.reform.fpl.service;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.google.common.collect.ImmutableMap;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.autoconfigure.jackson.JacksonAutoConfiguration;
+import org.springframework.boot.test.mock.mockito.MockBean;
+import org.springframework.test.context.ContextConfiguration;
+import org.springframework.test.context.junit.jupiter.SpringExtension;
+import uk.gov.hmcts.reform.ccd.client.model.CaseDetails;
+import uk.gov.hmcts.reform.fpl.config.DefaultEmailLookupConfiguration;
+import uk.gov.hmcts.reform.fpl.config.LocalAuthorityEmailLookupConfiguration;
+import uk.gov.hmcts.reform.fpl.model.Solicitor;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.mockito.BDDMockito.given;
+
+@ExtendWith(SpringExtension.class)
+@ContextConfiguration(classes = {JacksonAutoConfiguration.class, LocalAuthorityEmailLookupConfiguration.class,
+    InboxLookupService.class, DefaultEmailLookupConfiguration.class})
+public class InboxLookupServiceTest {
+
+    @MockBean
+    private LocalAuthorityEmailLookupConfiguration localAuthorityEmailLookupConfiguration;
+
+    @MockBean
+    private DefaultEmailLookupConfiguration defaultEmailLookupConfiguration;
+
+    @Autowired
+    private ObjectMapper mapper;
+
+    private InboxLookupService inboxLookupService;
+
+    private static final String LOCAL_AUTHORITY_CODE = "example";
+    private static final String LOCAL_AUTHORITY_EMAIL_ADDRESS = "FamilyPublicLaw+sa@gmail.com";
+    private static final String SOLICITOR_EMAIL_ADDRESS = "FamilyPublicLaw+sa@gmail.com";
+    private static final String DEFAULT_EMAIL = "FamilyPublicLaw@gmail.com";
+
+    @BeforeEach
+    void setup() {
+        this.inboxLookupService =
+            new InboxLookupService(mapper,
+                localAuthorityEmailLookupConfiguration,
+                defaultEmailLookupConfiguration);
+    }
+
+    @Test
+    void shouldReturnLocalAuthorityEmailWhenEmailExist() {
+        CaseDetails caseDetails = buildCaseDetailsWithSolicitorEmail();
```

Comment:
confused by buildCaseDetailsWithSolicitorEmail but assertions around local authority email

Reply:
It is because same method has been used in below tests for code reusability. It was `buildCaseDetails` before but then on below test cases it was not clear where we are doing assertion around solicitor email. May be keeping `buildCaseDetails` will work with all. Please suggest.

Reply:
@swalker125 We agreed to revert back to more generic `buildCaseDetails`.

Reply:
`LOCAL_AUTHORITY_EMAIL_ADDRESS` and `SOLICITOR_EMAIL_ADDRESS` are the same. Are both of them needed? Would only having one (probably LA as I think a solicitor is an LA) solve this confusion if the method was then renamed `buildCaseDetailsWithLAEmail` or something like that.

Reply:
Changed it to `buildCaseDetails` for now.